src/main/java/io/stargate/sgv2/jsonapi/service/shredding/collections/DocumentShredder.java

```diff
@@ -62,26 +62,27 @@ public DocumentShredder(
   }
 
   /**
-   * Shreds a single JSON node into a {@link WritableShreddedDocument} representation.
+   * Test-only shred-method that does not require a {@link CommandContext} and passes in a
+   * hard-coded command name and indexing projector. This is only used for testing: ideally should
+   * be refactored away: named specifically to avoid confusion with the actual production code. MUST
+   * NOT be called by non-test code.
    *
-   * @param document {@link JsonNode} to shred.
-   * @return WritableShreddedDocument
+   * @param doc Document to shred.
+   * @return Shredded document
    */
-  public WritableShreddedDocument shred(JsonNode document) {
-    return shred(document, null);
-  }
-
-  public WritableShreddedDocument shred(JsonNode doc, UUID txId) {
-    // TODO - why does this have "testCommand" for a command name ? Should it be a constant ?
+  public WritableShreddedDocument testShred(JsonNode doc, UUID txId) {
```

Comment:
this feels pretty janky, can we atleast get a @VisibleForTests . Can we merge and then try to refactor this out ? It also uses the  CollectionSchemaObject.MISSING_WITH_LEXICAL which i commented on, the original MISSING constant was not a great idea -> if the collection is not found then we should not have an object for it.

Reply:
Yes, very gnarly. But I wanted to outline this instead of existing code effectively hiding it -- was/am worried regular code could have inadvertently called it.

We have about 80 tests calling the method but I'll do a follow-up issue to weed this out.

Reply:
On `CollectionSchemaObject`: that is used to check collection settings like indexing config, vector/lexical enabling. So tests unfortunately do need to pass it (or bigger refactoring needed to change how collection config is passed). But I created #1971 for clean up.